src/pages/registration/description_tab/vocabularies/VocabularyFields.tsx

```diff
@@ -0,0 +1,124 @@
+import { Button, Menu, MenuItem, Typography } from '@material-ui/core';
+import AddIcon from '@material-ui/icons/Add';
+import RemoveCircleIcon from '@material-ui/icons/RemoveCircle';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import styled from 'styled-components';
+import ConfirmDialog from '../../../../components/ConfirmDialog';
+import DangerButton from '../../../../components/DangerButton';
+import { HrcsActivityAutocomplete } from './HrcsActivityAutocomplete';
+import { HrcsCategoryAutocomplete } from './HrcsCategoryAutocomplete';
+import { dataTestId } from '../../../../utils/dataTestIds';
+
+const StyledAddButton = styled(Button)`
+  margin-top: 1rem;
+`;
+
+const StyledRemoveButton = styled(DangerButton)`
+  margin-top: 1rem;
+  @media (max-width: ${({ theme }) => theme.breakpoints.values.sm + 'px'}) {
+    margin-top: 0.5rem;
+  }
+`;
+
+const StyledVocabularyRow = styled.div`
+  display: grid;
+  grid-template-columns: 5fr 1fr;
+  column-gap: 2rem;
+
+  @media (max-width: ${({ theme }) => theme.breakpoints.values.sm + 'px'}) {
+    grid-template-columns: 1fr;
+  }
+`;
+
+interface VocabularyConfig {
+  [key: string]: { i18nKey: string; component: () => JSX.Element };
+}
+
+// Specify which vocabularies to show, and their i18n key and component
+const vocabularyConfig: VocabularyConfig = {
+  hrcsActivity: {
+    i18nKey: 'description.hrcs_activities',
+    component: HrcsActivityAutocomplete,
+  },
+  hrcsCategory: {
+    i18nKey: 'description.hrcs_categories',
+    component: HrcsCategoryAutocomplete,
+  },
+};
+const vocabularies = Object.keys(vocabularyConfig);
+
+export const VocabularyFields = () => {
+  const { t } = useTranslation('registration');
+  const [newVocabularyAnchor, setNewVocabularyAnchor] = useState<null | HTMLElement>(null);
+  const [visibleVocabularies, setVisibleVocabularies] = useState<string[]>([]);
+  const [vocabularyToRemove, setVocabularyToRemove] = useState('');
+
+  const addableVocabularies = vocabularies.filter((vocabulary) => !visibleVocabularies.includes(vocabulary));
+
+  return (
+    <>
+      {visibleVocabularies.map((vocabulary) => {
+        const VocabularyInputComponent = vocabularyConfig[vocabulary].component;
+        return (
+          <StyledVocabularyRow
+            key={vocabulary}
+            data-testid={dataTestId.registrationWizard.description.vocabularyRow(vocabulary)}>
```

Comment:
Enig i at det er like greit å legge testid her @Eiriknil ?
Da kan man finne tekstfelt og fjern-knapp med feks:
```
cy.get("vocabulary-row-hrcsActivity > input");
cy.get("vocabulary-row-hrcsActivity > button");
```
I stedet for at vi lager separate ider for hver input og knapp, siden de ofte har en lei tendens til å havne på feil nivå og/eller endre navn

Reply:
Enig